functional/FunctionalTest.java

```diff
@@ -3284,6 +3284,230 @@ public static void composeObject_test6() throws Exception {
+            null,
+            startTime,
+            null,
+            e.toString() + " >>> " + Arrays.toString(e.getStackTrace()));
+        throw e;
+      }
+    }
+  }
+
+  /** Test: disableObjectLegalHold(String bucketName, String objectName, String versionId) */
+  public static void disableObjectLegalHold_test() throws Exception {
+    if (!mintEnv) {
+      System.out.println(
+          "Test: disableObjectLegalHold(String bucketName, String objectName, String versionId)");
+    }
+    long startTime = System.currentTimeMillis();
+    String bucketName = getRandomName();
+    String objectName = getRandomName();
+    try {
+      client.makeBucket(bucketName, null, true);
+      try {
+        try (final InputStream is = new ContentInputStream(1 * KB)) {
+          client.putObject(bucketName, objectName, is, new PutObjectOptions(1 * KB, -1));
+        }
+        client.enableObjectLegalHold(bucketName, objectName, null);
+        client.disableObjectLegalHold(bucketName, objectName, null);
+        if (client.isObjectLegalHoldEnabled(bucketName, objectName, null)) {
+          throw new Exception("[FAILED] isObjectLegalHoldEnabled(): expected: false, got: true");
+        }
+      } finally {
+        client.removeObject(bucketName, objectName);
+        client.removeBucket(bucketName);
+      }
+      mintSuccessLog(
+          "disableObjectLegalHold(String bucketName, String objectName, String versionId)",
+          null,
+          startTime);
+    } catch (Exception e) {
+      ErrorResponse errorResponse = null;
+      if (e instanceof ErrorResponseException) {
+        ErrorResponseException exp = (ErrorResponseException) e;
+        errorResponse = exp.errorResponse();
+      }
+
+      // Ignore NotImplemented error
+      if (errorResponse != null && errorResponse.errorCode() == ErrorCode.NOT_IMPLEMENTED) {
+        mintIgnoredLog(
+            "disableObjectLegalHold(String bucketName, String objectName, String versionId)",
+            null,
+            startTime);
+      } else {
+        mintFailedLog(
+            "disableObjectLegalHold(String bucketName, String objectName, String versionId)",
+            null,
+            startTime,
+            null,
+            e.toString() + " >>> " + Arrays.toString(e.getStackTrace()));
+        throw e;
+      }
+    }
+  }
+
+  /** Test: setDefaultRetention(String bucketName). */
+  public static void setDefaultRetention_test() throws Exception {
+    if (!mintEnv) {
+      System.out.println("Test: setDefaultRetention(String bucketName)");
+    }
+
+    long startTime = System.currentTimeMillis();
+    String bucketName = getRandomName();
+
+    try {
+      client.makeBucket(bucketName, null, true);
+      try {
+        ObjectLockConfiguration config =
+            new ObjectLockConfiguration(RetentionMode.COMPLIANCE, new RetentionDurationDays(10));
+        client.setDefaultRetention(bucketName, config);
```

Comment:
Here also we can attempt an upload and confirm the uploaded object can't be deleted.

Reply:
If we do so, we can't do clean up of the test.

Reply:
IMO Testing the MinIO Server's behaviour is more important, since this test will become part of Mint. It will be used to validate MinIO Server behaviour. We generally test against Play or Local MinIO instances - in both cases cleanup is not that important